department-of-veterans-affairs/caseflow
# JHoang/APPEALS-35647

Resolves [APPEALS-35647](https://jira.devops.va.gov/browse/APPEALS-35647)

# Description
Give attorneys a new action dropdown "Cancel task and return to Specialty Case Team".
When attorneys submit this task, the `AttorneyTask` + the parent task(`JudgeTask`) will be canceled and the `SpecialtyCaseTeamAssignTask` will be re-opened with a status of `in_progress`. The result of this will move the appeal out of the attorney's queue and back into the Specialty Case Team's queue.

## Acceptance Criteria

## Testing Plan
1. We must first assign an SCT appeal to an attorney first. This can be done either two ways so far:
  -  Rails Console (I used this method)
    - Lets pick an attorney that does not have any cases assigned. I chose BVABCASPER. `attorney = User.find_by_css_id("BVABCASPER")`
    - We don't care about the judge so lets create the SCT appeal now with the specified attorney.`FactoryBot.create(:specialty_case_team_assign_task, :completed, associated_attorney: attorney)`
  - SCT Bulk Assign page
    - Login as an SCT Coordinator (SCTUser or SCTAdmin)
    - Click on "Your Queue"
    - Click on "Switch views"
    - Click on "Assign SCT Appeals team cases"
    - Check a checkbox for any of these appeals.
    - Click on the dropdown and pick any attorney you like.
    - Click on the "Assign _n_ cases" button
2. Go to the case detail's page for that newly created appeal. You should be able to find the link in the attorney's queue.
3. Click on the action dropdown
4. You should see a new option labeled "Cancel task and return to Specialty Case Team"
<img width="1432" alt="Screenshot 2024-02-16 at 6 46 08 PM" src="https://github.com/department-of-veterans-affairs/caseflow/assets/111081469/6653affc-ee4e-4309-b680-7734eae92de9">
5. Click on the new option and the Return to SCT Modal should appear.
<img width="489" alt="Screenshot 2024-02-16 at 6 47 43 PM" src="https://github.com/department-of-veterans-affairs/caseflow/assets/111081469/d677fd45-46dc-4e60-b1ba-b646fc5f142f">

6. Observe that the "Return" button is disabled.
7. Fill in the text field with a reasoning why the case should be returned to SCT.
8. Observe that the "Return" button is enabled.
9. Click on the "Return" button.
10. You should be redirected to the attorney's queue.
11. A success banner should appear indicating that the appeal has been moved to the SCT Queue. 
<img width="1470" alt="Screenshot 2024-02-16 at 6 50 01 PM" src="https://github.com/department-of-veterans-affairs/caseflow/assets/111081469/a4df62ae-7693-4ea8-acf5-87550499af42">

12. Observe that the appeal is no longer in the attorney's queue.
13. Now login as an SCT Coordinator (SCTUser or SCTAdmin)
14. Click on "Your Queue"
15. Click on "Switch views"
16. Click on "Specialty Case Team team cases"
17. If you are not at the Action Required tab, click on that tab.
18. Observe that the SCT appeal appears in this tab.


# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
